### ISSUE TITLE
[ Flow control ]

### ISSUE THREAD

**Issue by jerome83136** [rundeck/rundeck#2298]
Hi,

I'm using the latest version of Rundeck.

I'm trying to set up a job workflow with "optional" jobs.
All my jobs have dynamic filters and, depending of the user input, some jobs will have no nodes to run on because dynamic filter's values will not match.

I am trying to use the error handler to continue the workflow when a job has the "NoMatchedNodes: No nodes matched for the filters" error.
But when a job runs, I still want it to fail if there is another problem. I just want it to not fail when the error "NoMatchedNodes: No nodes matched for the filters" occurs.

Can I do this and, if it is possible, how ?

Thank you for your help.
Regards
Jérôme

**Comment by gschueler** [comment]
yes, you should be able to add an error handler Command step like: ` test "${result.reason}" == "NoMatchedNodes" && echo "no nodes matched, skipping" ` and make sure "keep going on success" is checked.  The `${result.reason}` value will indicate the NoMatchedNodes failure and allow you to continue.
